oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/ClientSecretBasicAuthenticationConverter.java

```diff
@@ -86,8 +90,10 @@ public Authentication convert(HttpServletRequest request) {
 			throw new OAuth2AuthenticationException(new OAuth2Error(OAuth2ErrorCodes.INVALID_REQUEST), ex);
 		}
 
-		return new OAuth2ClientAuthenticationToken(clientID, clientSecret, ClientAuthenticationMethod.BASIC,
+		OAuth2ClientAuthenticationToken oAuth2ClientAuthenticationToken = new OAuth2ClientAuthenticationToken(clientID, clientSecret, ClientAuthenticationMethod.BASIC,
 				extractAdditionalParameters(request));
+		oAuth2ClientAuthenticationToken.setDetails(authenticationDetailsSource.buildDetails(request));
```

Comment:
Instead of updating `ClientSecretBasicAuthenticationConverter`, `ClientSecretPostAuthenticationConverter`, etc. we should update `OAuth2ClientAuthenticationFilter.doFilterInternal()` as such:

```java
Authentication authenticationRequest = this.authenticationConverter.convert(request);
if (authenticationRequest instanceof AbstractAuthenticationToken) {
	((AbstractAuthenticationToken) authenticationRequest).setDetails(
			authenticationDetailsSource.buildDetails(request));
}
```

Please apply the same to `OAuth2TokenEndpointFilter.doFilterInternal()`.

Ultimately, the changes should only be applied in the `Filter`'s.

Reply:
Thanks for recommending these changes. I've addressed them, hopefully according to your comment.